jdk/src/share/bin/java.c

```diff
@@ -297,29 +335,17 @@ JLI_Launch(int argc, char ** argv,              /* main argc, argc */
         SetClassPath(cpath);
     }
 
+    /* Process -Xmso to set the main thread stack size. May be overridden by
+     * a later command line option.
+     */
     {
-        /* Process -Xmso in the OPENJ9_JAVA_OPTIONS environment variable to
-         * set the main thread stack size. May be overridden by a later command
-         * line option.
-         */
-        JLI_List openj9Args = JLI_List_new(8); /* 8 is arbitrary */
-        if (JLI_ParseOpenJ9ArgsFromEnvVar(openj9Args, "OPENJ9_JAVA_OPTIONS")) {
-            size_t i = openj9Args->size;
-            while (i > 0) {
-                i -= 1;
-                if (JLI_StrCCmp(openj9Args->elements[i], "-Xmso") == 0) {
-                    jlong tmp = 0;
-                    if (parse_size(openj9Args->elements[i] + 5, &tmp)) {
-                        threadStackSize = tmp;
-                        if (threadStackSize > 0 && threadStackSize < (jlong)STACK_SIZE_MINIMUM) {
-                            threadStackSize = STACK_SIZE_MINIMUM;
-                        }
-                    }
-                    break;
-                }
-            }
-            JLI_List_free(openj9Args);
-        }
+#define OPTIONS_DEFAULT_PATH "/lib/" LIBARCHNAME "/default/options.default"
```

Comment:
It occurs to me that the `default` directory doesn't exist in some configurations (e.g. `--with-noncompressedrefs=no --with-mixedrefs=no` yields only the `compressedrefs` directory):
```
$ grep OPENJ9_LIBS_SUBDIR custom-spec.gmk 
OPENJ9_LIBS_SUBDIR      := compressedrefs
```
All Semeru configurations actually distributed have the `default` directory, so maybe we can ignore this?

Reply:
A possible solution: change the definition of `OPTIONS_DEFAULT_PATH`:
```c
#define OPTIONS_DEFAULT_PATH "/lib/" LIBARCHNAME "/" OPENJ9_LIBS_SUBDIR "/options.default"
```
and add to `custom-spec.gmk.in`
```
OPENJDK_TARGET_CPU_JLI_CFLAGS += -DOPENJ9_LIBS_SUBDIR='"@OPENJ9_LIBS_SUBDIR@"'
```

Reply:
I missed your earlier comment. I'm wondering if we should remove support for `--with-mixedrefs=no`, but I'll implement your suggestion. Also Windows isn't correct, since the file is in `bin\default`.

Reply:
Updated.

Reply:
I created https://github.com/eclipse-openj9/openj9/issues/18124 for removal of the `--with-mixedrefs` configuration option.